Return packed array

hpinkos opened the pull request:
This make all `pack` functions return the packed array.  This helpful if I want to do something like 

``` javascript
var myObject = {
   position: Cartesian3.pack(position, new Array(3));
} 
```

I also think it's all around more intuitive.  I know we do this a lot, but to me this syntax is weird
``` javascript
Cartesian3.pack(position, array);
```

And this makes much more sense 
``` javascript
array = Cartesian3.pack(position, array);
```

pjcozzi commented:
This is also well-aligned with math functions like `Cartesian3.add`.

Can you update reference doc, update CHANGES.md, and add full unit tests?

hpinkos commented:
@pjcozzi updated CHANGES.md and added doc

I think all classes that have `pack` use `createPackableSpecs` to create their pack specs.  Since I check for the array to be returned there, I think that should cover all of the functions I changed

mramato commented on Specs/createPackableSpecs.js:
```diff
@@ -14,7 +14,8 @@ define([
             var packedArray = [];
-            packable.pack(instance, packedArray);
+            packedArray = packable.pack(instance, packedArray);
+            expect(packedArray).toBeDefined();
```
Rather than re-assigned packagedArray, create a new variable called `returnedArray` and do `expect(returnedAraray).toBe(packedArray)` in order to verify that's what is actually returned.

mramato commented:
Except for that one comment, this is fine with me.  Thanks, @hpinkos!

hpinkos commented:
Thanks @mramato!  Updated
